src/pages/index.js

```diff
@@ -2,56 +2,117 @@ import FormValidator from "../components/FormValidator.js";
+import PopupWithConfirmation from "../components/PopupWithConfirmation.js";
 import Section from "../components/Section.js";
 import UserInfo from "../components/UserInfo.js";
-import { initialCards, config } from "../utils/constants.js";
+import { config } from "../utils/constants.js";
+import Api from "../components/Api.js";
 import "./index.css";
 
 //wrappers
 const profileEditModal = document.querySelector("#profile-edit-modal");
 const profileEditForm = profileEditModal.querySelector(".modal__form");
 const addCardModal = document.querySelector("#add-card-modal");
 const addCardForm = addCardModal.querySelector(".modal__form");
+const cardSelector = "#card-template";
+const avatarForm = document.querySelector("#avatar-form");
 
 //add card variables
 const addCardBtn = document.querySelector(".profile__add-button");
 
+
+//profile avatar form
+
+const profileAvatarBtn = document.querySelector(".profile__avatar-btn");
+
 // Buttons + DOM nodes
 const profileEditBtn = document.querySelector(".profile__edit-button");
 
-
 // Form data
 const nameInput = document.querySelector(".modal__input_type_name");
 const descriptionInput = document.querySelector(
   ".modal__input_type_description"
 );
 
-
-//Objects
-const previewImage = new PopupWithImage({
-  popupSelector: "#modal__preview-image",
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "c06cb1ea-bab9-48c7-8798-47ebeebe8696",
+    "Content-Type": "application/json",
+  },
 });
-const cardSelector = "#card-template";
 
-previewImage._setEventListeners();
-function createCard(data) {
-  const card = new Card(data, cardSelector, (name, link) => {
-    previewImage.open({ link, name });
+let section;
+api
+  .getInitialCards()
+  .then((cards) => {
+    section = new Section(
+      {
+        items: cards,
+        renderer: createCard,
+      },
+      ".cards__list"
+    );
+
+    section.renderItems();
+  })
+  .catch((err) => {
+    console.error(err);
   });
+
+function createCard(data) {
+  const card = new Card(
+    data,
+    cardSelector,
+    handleDeleteClick,
+    handleLikeClick,
+    (name, link) => {
+      previewImage.open({ link, name });
+    }
+  );
   return card.getView();
 }
 
-const section = new Section(
-  { items: initialCards, renderer: createCard },
-  ".cards__list"
-);
-section.renderItems();
-
 const userInfo = new UserInfo({
   nameSelector: ".profile__name",
   jobSelector: ".profile__description",
+  avatarSelector: ".profile__picture",
+});
+
+api.getInitialInfo().then((userData) => {
+  userInfo.setAvatarImg(userData.avatar);
+  userInfo.setUserInfo({
+    name: userData.name,
+    description: userData.about,
+  });
 });
 
+function handleDeleteClick(card) {
+  deleteCardConfirm.open();
+  deleteCardConfirm.setSubmitAction(() => {
+   
+    api
+      .deleteCard(card._cardId)
+      .then(() => {
+        deleteCardConfirm.close();
+        card._handleDeleteIcon();
```

Comment:
`_handleDeleteIcon`

Please, delete the **low dash** (`underscore`) from methods if you want to use them `outside` a class because the `low dash` shows a **private** method which is used only inside the class.


**Please, correct it everywhere**